### ISSUE TITLE
Feasible benchmark of the trainers

### ISSUE THREAD

**Issue by accosmin** [accosmin/nano#57]
Should increase the number of samples to more consistent results.
Should investigate a fairer and *faster* minibatch setting.

Ideally this benchmark should not take more than 2-3 hours on a decent 4+ core system.

**Comment by accosmin** [comment]
Currently a minibatch run takes around 15 times more than a batch one. This difference should be greatly reduced.

**Comment by accosmin** [comment]
Improvements:
* randomly shuffle ONCE the whole training dataset and then process the samples in that order.
* instead of tuning the batch size, tune a splitting factor (1/256, 1/128 ... 1/16).
